WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -1,23 +1,27 @@
 import ScreenObject
 import XCTest
 
-public class SingleProductScreen: ScreenObject {
+public final class SingleProductScreen: ScreenObject {
 
     static var isVisible: Bool {
         guard let screen = try? SingleProductScreen() else { return false }
         return screen.isLoaded && screen.expectedElement.isHittable
     }
 
-    init(app: XCUIApplication = XCUIApplication()) throws {
+    public init(app: XCUIApplication = XCUIApplication()) throws {
         try super.init(
-            expectedElementGetters: [ { $0.buttons["edit-product-more-options-button"] }],
+            expectedElementGetters: [ {$0.buttons["edit-product-more-options-button"]} ],
             app: app
         )
     }
 
     @discardableResult
-    func goBackToProductList() throws -> ProductsScreen {
+    public func goBackToProductList() throws -> ProductsScreen {
         navBackButton.tap()
         return try ProductsScreen()
     }
+
+    public func verifyProductNameOnSingleProductScreen(name: String) {
+        XCTAssert(app.textViews[name].exists, "Product name does not exist!")
+    }
```

Comment:
The dump of objects hierarchy (via `print(app.debugDescription)`) shows that there are some hierarchy specifics that will help us to assert more than the Product name:

<img width="699" alt="Screenshot 2021-12-15 at 00 08 00" src="https://user-images.githubusercontent.com/73365754/146087708-078bf9a4-3676-4d87-afde-b9796583f86c.png">

Highlighted is an example of the cell where a static `StaticText` (green `Inventory`) can be used to locate a dynamic static text (blue).

To be honest, since this is a single product screen, I'm not sure if there's a lot of sense in checking that both `Inventory` and `<stock_status>` really belong to the same cell (but it makes sense in products list). I think it's fine to check that the elements are just there.

^ Same can be done for Price, product type and number of reviews.

Reply:
Alright, for single product screen we can add these assertions. I'll include stock_status, price, product type and number of reviews. I'll see how to best do this, and if it's too much changes may end up in a separate PR (together with the extra checks on Products screen)